pkg/gui/controllers/helpers/commits_helper.go

```diff
@@ -0,0 +1,140 @@
+package helpers
+
+import (
+	"strings"
+
+	"github.com/jesseduffield/lazygit/pkg/gui/context"
+	"github.com/jesseduffield/lazygit/pkg/gui/types"
+)
+
+type ICommitsHelper interface {
+	UpdateCommitPanelView(message string, description string)
+}
+
+type CommitsHelper struct {
+	c *types.HelperCommon
+
+	model                *types.Model
+	contexts             *context.ContextTree
+	getCommitMessage     func() string
+	getCommitDescription func() string
+	setCommitMessage     func(string)
+	setCommitDescription func(string)
+	renderCommitLength   func()
+}
+
+var _ ICommitsHelper = &CommitsHelper{}
+
+func NewCommitsHelper(
+	c *types.HelperCommon,
+	model *types.Model,
+	contexts *context.ContextTree,
+	getCommitMessage func() string,
+	getCommitDescription func() string,
+	setCommitMessage func(string),
+	setCommitDescription func(string),
+	renderCommitLength func(),
+) *CommitsHelper {
+	return &CommitsHelper{
+		c:                    c,
+		model:                model,
+		contexts:             contexts,
+		getCommitMessage:     getCommitMessage,
+		getCommitDescription: getCommitDescription,
+		setCommitMessage:     setCommitMessage,
+		setCommitDescription: setCommitDescription,
+		renderCommitLength:   renderCommitLength,
+	}
+}
+
+func (self *CommitsHelper) SplitCommitMessageAndDescription(message string) (string, string) {
+	msg, description, found := strings.Cut(message, "\n\n")
+	if !found {
+		return message, ""
+	}
+	return msg, description
+}
+
+func (self *CommitsHelper) SetMessageAndDescriptionInView(message string, description string) {
+	self.setCommitMessage(message)
+	self.setCommitDescription(description)
+	self.renderCommitLength()
+}
+
+func (self *CommitsHelper) UpdateCommitPanelView(message string, description string) {
+	// first try the passed in message, if not fallback to context -> view in that order
```

Comment:
Ideally I would to simply set whatever message and description comes through the args. From what I've seen, when a commit fails, the commit message context is cleared at some point, is that correct? Since that's the case, we need to fallback to render whatever is in the view, because that's not cleared. It would be much simpler if the context wasn't cleared when there is an error though :thinking:

Reply:
I still prefer to the clear the context whenever we need to close is given that depending on what's in the view always ends up causing issues down the line.

If the commit fails, I would not expect the message to go away. Is that currently happening?